service/matching/ack_manager.go

```diff
@@ -144,7 +146,10 @@ func (m *ackManager) completeTask(taskID int64) int64 {
 		}
 		m.ackLevel = min.(int64)
 		m.outstandingTasks.Remove(min)
+		// reducing our backlog since a task gets acked
```

Comment:
nit: I would maybe use "reduce" conjugation instead of "reducing." Possibly: "decrement backlog as each task gets acked"